pkg/alert/commands.go

```diff
@@ -318,3 +316,187 @@ func NewDeleteAlertSettingConditionCommand() *cobra.Command {
 
 	return cmd
 }
+
+//NewGetAlertChannelsCommand creates the `alertchannels` command
+func NewGetAlertChannelsCommand() *cobra.Command {
+	var (
+		page         int
+		pageSize     int
+		sortField    string
+		sortOrder    string
+		templateName string
+		channelName  string
+		details      bool
+	)
+
+	cmd := &cobra.Command{
+		Use:              "alertchannels",
+		Short:            "Print the registered alert channels",
+		Example:          `alertchannels --page=1 --pageSize=10 --sortField="name" --sortOrder="asc" --templateName="test" --channelName="slack" --details`,
+		TraverseChildren: true,
+		SilenceErrors:    true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var (
+				alertchannels            api.AlertChannelResponse
+				alertchannelsWithDetails api.AlertChannelResponseWithDetails
+				err                      error
+			)
+			if details {
+				alertchannelsWithDetails, err = config.Client.GetAlertChannelsWithDetails(page, pageSize, sortField, sortOrder, templateName, channelName)
+			} else {
+				alertchannels, err = config.Client.GetAlertChannels(page, pageSize, sortField, sortOrder, templateName, channelName)
+			}
+			if err != nil {
+				golog.Errorf("Failed to retrieve alert channels. [%s]", err.Error())
+				return err
+			}
+
+			if !details {
```

Comment:
also this seems to me a duplicate of what you do with `else`. in else just `return`. In general fail fast, return fast.